test/test_cli.py

```diff
@@ -242,10 +244,325 @@ def test_should_return_None_when_no_root_property(self):
         args = lambda: None
 
         # when
-        res = read_root(args)
+        res = read_project_package(args)
 
         # then
         self.assertEqual(None, res)
 
     def assert_started_jobs(self, ids):
         self.assertEqual(ids, import_module("test_module.Unused1").started_jobs)
+
+
+    def test_should_decode_version_number_from_file_name(self):
+        # given
+        f_release = self._touch_file('image-0.1.0.tar')
+        f_dev = self._touch_file('image-0.3.0.dev-4b45b638.tar')
+
+        # expect
+        self.assertEqual(_decode_version_number_from_file_name(f_release), '0.1.0')
+        self.assertEqual(_decode_version_number_from_file_name(f_dev), '0.3.0.dev-4b45b638')
+
+        f_release.unlink()
+        f_dev.unlink()
+
+    def test_should_raise_error_when_given_image_file_is_not_tar(self):
```

Comment:
what if file will be .tar.gz?

Reply:
won't be :) we genmerate it by ourselves, somewhere in the build process

Reply:
👍